packages/contracts/script/deployer/DeployVaults.s.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import { Script } from "forge-std/Script.sol";
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import { CredbullFixedYieldVaultFactory } from "@credbull/CredbullFixedYieldVaultFactory.sol";
+import { CredbullUpsideVaultFactory } from "@credbull/CredbullUpsideVaultFactory.sol";
+import { CredbullWhiteListProvider } from "@credbull/CredbullWhiteListProvider.sol";
+import { Vault } from "@credbull/vault/Vault.sol";
+
+import { SimpleUSDC } from "@test/test/token/SimpleUSDC.t.sol";
+import { SimpleToken } from "@test/test/token/SimpleToken.t.sol";
+import { SimpleVault } from "@test/test/vault/SimpleVault.t.sol";
+
+import { VaultsConfigured } from "./Configured.s.sol";
+import { VaultsSupportConfigured } from "./Configured.s.sol";
+
+/// @notice The [Script] used to deploy the Credbull Vaults Distribution.
+contract DeployVaults is Script, VaultsConfigured {
+    /**
+     * @notice The `forge script` invocation entrypoint, this deploys the Vaults Distribution Unit and, optionally,
+     *  the Vaults Support Distribution Unit, if deployment is enabled.
+     * @dev For command line invocation, the return values are ignored.
+     *
+     * @return fixedYieldVaultFactory The deployed [CredbullFixedYieldVaultFactory].
+     * @return upsideVaultFactory The deployed [CredbullUpsideVaultFactory].
+     * @return whiteListProvider The deployed [CredbullWhiteListProvider].
+     */
+    function run()
```

Comment:
I simplified these, removing all trace of 'test mode' or 'deploy checking'. Why? 
1. In a deployment setting, there is already a deployment check step, so having it IN the scripts is redundant.
2. In test, we don't actually care about existing contracts (`setUp()` re-set the chain each time) and must simply deploy.